### ISSUE TITLE
Possible to use SHA1 instead of MD5?

### ISSUE THREAD

**Issue by alexzherdev** [rdy/fixture_builder#55]
Probably a long shot, but we're trying to use fixture_builder with OpenSSL running in [FIPS mode](https://www.openssl.org/docs/fips.html). Under this mode, MD5 digest is not available, but it's being used to calculate file hashes [here](https://github.com/rdy/fixture_builder/blob/4252b2a9c4b4a8efe504d3373c9349b91fc51a8e/lib/fixture_builder/configuration.rb#L119).

SHA1 would work for FIPS purposes, but I understand that changing the algorithm would make users unnecessarily regenerate their fixtures. Any chance this can be made customizable?

**Comment by thewoolleyman** [comment]
If you want to submit a PR with tests that adds this as an option you can pass in to `FixtureBuilder.configure()` then that seems reasonable.

Would this require a new gem dependency, or can it be done with the ruby standard lib?  The latter is preferable.

**Comment by alexzherdev** [comment]
No, `Digest::SHA1` is part of the standard library just like `Digest::MD5`.
I think it's probably not worth it to overcomplicate this—a boolean option `use_sha1_digests` should be enough, like [in Rails](https://github.com/rails/rails/pull/31651)? I doubt anyone would ever want some third option.

**Comment by thewoolleyman** [comment]
Sure, a boolean option to `FixtureBuilder.configure()` seems fine, with the default being the existing behavior.  

Also great that it's in the standard library.  Just make sure it exists, and works, in all our supported versions of Ruby (which I'm not sure if we actually specify anywhere, but at least >= 2.0).

**Comment by thewoolleyman** [comment]
@alexzherdev I just released `0.5.3.rc1` with this change.  Give it a try and let me know if it works for you!

**Comment by alexzherdev** [comment]
> Just make sure it exists, and works, in all our supported versions of Ruby (which I'm not sure if we actually specify anywhere, but at least >= 2.0).

Yup, `Digest::SHA1` has been available in Ruby for ages, at least since 1.8.

> @alexzherdev I just released `0.5.3.rc1` with this change. Give it a try and let me know if it works for you!

Thanks! And sorry, it doesn't work, I should have actually verified this against my project 🤦‍♂ .
`file_hashes` is first called in `Configuration`'s constructor, when the configure block hasn't had a chance to get called. It looks to me that line is necessary.
https://github.com/rdy/fixture_builder/blob/88d2469373d801a8c509ee0f3f489c85c39a7030/lib/fixture_builder/configuration.rb#L25-L29

With `opts={}` already here, we could pass a flag as an option instead. (I did verify that this works)
```rb
FixtureBuilder.configure(use_sha1_digests: true) do ... end
```
Unless that looks too ugly to you.

**Comment by thewoolleyman** [comment]
ok, just open a new PR.

**Comment by alexzherdev** [comment]
Do you think it's worth documenting in the README?

**Comment by thewoolleyman** [comment]
If you want to throw a blurb in the readme, that's fine.

**Comment by thewoolleyman** [comment]
OK, I merged your changes and pushed as 0.5.3.rc2.  Give it another try, please.

**Comment by alexzherdev** [comment]
Works great, thanks for your assistance!
